### PR TITLE
Adds option to watch specific namespaces, some refactoring

### LINKED ISSUE #27: Option to specify namespaces to watch for ingresses
In some environments it's required for the ingress controller to watch for ingresses only in specific namespaces rather than all of them, e.g. for isolation purposes in multi-tenant clusters.

Currently watching all the namespaces is hardcoded, should instead have an option to specify the namespaces (like `--watch-namespace` in nginx ingress controller) for this use case.

### PR DESCRIPTION
**Pull request by excieve**
Watching resources only in some specified namespaces is frequently required in order to provide multi-tenancy or limit the reach of ingress controller for security purposes. So there's now a config option to provide a list of namespaces. In case none are provided it assumes all namespaces must be watched.

This also does some refactoring of how resources are watched, making use of API informers for ingresses and pods. It also makes sure caches get synchronised when starting the informers.

Since Kubernetes 1.14 ingresses are now part of the "networking" API instead of "extensions". While there's deprecation period, this change introduces support for both depending on the k8s version currently running.

Minor fixes and improvements added.

Fixes #27.

### CODE REVIEW

File: ingress/ingress.go [hunk 1]
```diff
@@ -16,37 +16,69 @@ import (
...
-	cfg               *Config
-	client            *kubernetes.Clientset
-	store             cache.Store
-	ingressController cache.Controller
-	podController     cache.Controller
-	stopCh            chan struct{}
+	cfg                 *Config
+	client              *kubernetes.Clientset
+	stopCh              chan struct{}
+	factories           map[string]informers.SharedInformerFactory
```
Comment (joshblakeley) [inline ingress/ingress.go]: I feel like calling the field `factories` is a bit non-idiomatic

Comment (excieve) [reply ingress/ingress.go]: Well, it is what it is :man_shrugging: What would you suggest instead?

Comment (joshblakeley) [reply ingress/ingress.go]: Oh boy yeah lets leave it >.<

**Comment by excieve** [comment]
Thanks for your review @joshblakeley!

**Pull request merged**